library/Icingadb/Widget/HostListItemDetailed.php

```diff
@@ -4,11 +4,78 @@
 
 namespace Icinga\Module\Icingadb\Widget;
 
+use Icinga\Module\Icingadb\Common\ListItemDetailedLayout;
 use Icinga\Module\Icingadb\Compat\CompatPluginOutput;
+use Icinga\Module\Icingadb\Util\PerfdataSet;
 use ipl\Html\BaseHtmlElement;
+use ipl\Html\Html;
+use ipl\Html\HtmlElement;
+use ipl\Html\HtmlString;
+use ipl\Web\Widget\Icon;
+use ipl\Web\Widget\StateBall;
 
-class HostListItemDetailed extends HostListItem
+class HostListItemDetailed extends BaseHostListItem
 {
+    use ListItemDetailedLayout;
+
+    protected $pie;
+
+    protected function getStateBallSize()
+    {
+        return StateBall::SIZE_LARGE;
+    }
+
+    protected function assembleFooter(BaseHtmlElement $footer)
+    {
+        $this->pie = HtmlString::create('');
+
+        $statusIcons = Html::tag('div', ['class' => 'status-icons']);
+
+        // ToDo(fs): Get `has_comments` from database
+        if ($this->item->comment->limit(1)->execute()->hasResult()) {
+            $statusIcons->add(new Icon('comments', ['title' => t('This item has been commented')]));
+        }
+
+        if (! $this->item->notifications_enabled) {
+            $statusIcons->add(new Icon('bell-slash', ['title' => t('Notifications disabled')]));
+        }
+
+        if (! $this->item->active_checks_enabled) {
+            $statusIcons->add(new Icon('eye-slash', ['title' => t('Active checks disabled')]));
+        }
+
+        $performanceData = Html::tag('div', ['class' => 'performance-data']);
+        if ($this->item->state->performance_data) {
+            $pieChartData = PerfdataSet::fromString($this->item->state->performance_data)->asArray();
+
+            $pies = [];
+            foreach ($pieChartData as $i => $perfdata) {
+                if ($perfdata->isVisualizable()) {
+                    $pies[] = $perfdata->asInlinePie()->render();
+                }
+
+                // Check if number of visualizable pie charts is larger than PIE_CHART_LIMIT
+                if (count($pies) > ServiceListItemDetailed::PIE_CHART_LIMIT) {
+                    break;
+                }
+            }
+
+            foreach ($pies as $i => $pie) {
+                if (
+                    $i > ServiceListItemDetailed::PIE_CHART_LIMIT - 2
+                    && count($pies) > ServiceListItemDetailed::PIE_CHART_LIMIT
```

Comment:
Please remove that. It doesn't make much sense to me to calculate the count on very iteration step. It's also unnecessary in my opinion, the first check should suffice.